Tools/CISupport/Shared/de-duper

```diff
@@ -114,23 +120,31 @@ def main(args):
             if os.stat(full_path).st_nlink > 1:
                 continue
             sha_value = sha1(full_path)
-            if sha_value in existing_links:
+            sha_path = os.path.join(args.storage, f'{sha_value}.png')
+            assert sha_path != full_path
+            if os.path.isfile(sha_path):
                 if args.dry_run:
                     print(f'[dry-run] Linking {full_path} to {sha_value}.png')
+                elif os.stat(sha_path).st_nlink >= max_links - 1:
```

Comment:
Gave this implementation a try, when combined with checking for file existence, it works quite well.

I had originally tried this. But it doesn't work well when paired with the old code which attempts to keep track of which images are in storage in program memory instead of checking disk each time. When we completely abandon in-memory representations and rely on the filesystem entirely, this works.